### DISCUSSION TITLE
Use comments to document/describe enum values

### DISCUSSION THREAD

**Discussion by m4tthumphrey** [dedoc/scramble#611] [Ideas]
As discussed in #584, it would be nice to be able to document Enum values. I was wondering if we could do something like this:

```
enum OrderStatus: int
{
    /**
     * Pending payment
     */
    case PENDING_PAYMENT = 1;

    /**
     * Paid
     */
    case PAID = 2;

    /**
     * Cancellec
     */
    case CANCELLED = 3;

    /**
     * Refunded
     */
    case REFUNDED = 4;

    /**
     * Error
     */
    case ERROR = 5;

    /**
     * Draft
     */
    case DRAFT = 6;

    /**
     * Intent
     */
    case INTENT = 7;
}
```

Which would then produce a format mentioned in the above issue. Thoughts?

**Comment by benascbr** [comment]
:+1: :pray: :prayer_beads:

**Comment by benascbr** [comment]
skibidi

**Comment by baralogi** [comment]
Hi i try to write `Rule::in(YourEnumClass::cases())` and working

**Reply by benascbr** [reply]
Please explain more.

**Reply by shavonn** [reply]
> Please explain more.

I think this response was advising that you add a rule for the column in your validation requiring it be the enum. Then it can glean the enum application from there.

I've achieved this by casting columns to enums in my model.

**Reply by baralogi** [reply]
hi output for YourEnumClass::cases() is array, example i have status enum class, later it will be like this `['PAID', 'UNPAID', 'PENDING']` 

on validation `Rule::in` you must input array as paramter... did you get in?
